src/file-uploader/file-uploader.js

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+
+import React from 'react';
+import checkFileType from 'attr-accept';
+
+import {Block} from '../block/index.js';
+import {
+  Root,
+  ContentContainer,
+  FileDragAndDrop,
+  FileSelectButton,
+  Input,
+} from './styled-components.js';
+
+import type {PropsT} from './types.js';
+
+export default class FileUploader extends React.Component<PropsT, any> {
+  inputRef: {current: ?HTMLInputElement};
+
+  constructor(props: PropsT) {
+    super(props);
+
+    this.inputRef = React.createRef();
+    this.state = {isDragActive: false, acceptedFiles: [], rejectedFiles: []};
+  }
+
+  componentDidMount() {
+    window.addEventListener('dragover', this.onWindowDragOverOrDrop);
+    window.addEventListener('drop', this.onWindowDragOverOrDrop);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('dragover', this.onWindowDragOverOrDrop);
+    window.removeEventListener('drop', this.onWindowDragOverOrDrop);
+  }
+
+  updateFiles = (files: File[], event: SyntheticEvent<HTMLElement>) => {
+    if (this.props.accept) {
+      let nextFiles = {acceptedFiles: [], rejectedFiles: []};
+      nextFiles = files.reduce((acc, file) => {
+        if (checkFileType(file, this.props.accept)) {
+          acc.acceptedFiles.push(file);
+        } else {
+          acc.rejectedFiles.push(file);
+        }
+        return acc;
+      }, nextFiles);
+
+      if (this.props.onChange) {
+        this.props.onChange({...nextFiles, event});
+      }
+
+      this.setState({
+        acceptedFiles: [
+          ...this.state.acceptedFiles,
+          ...nextFiles.acceptedFiles,
+        ],
+        rejectedFiles: [
```

Comment:
Why do we need to maintain the list of rejected files?

Reply:
this component simply wraps the `react-dropzone` package, so re-using that api